script.js

```diff
@@ -1,4 +1,447 @@
+// ____________________  Global Variables ________________________
+
+// L1 Game Mode variables to choose which game user wants to play
+var L1_PICK_GAME_MODE = "Choose your game";
+var L1_NORMAL_MODE = "Normal Mode";
+var L1_LOWEST_MODE = "Lowest Win";
+var L1_AUTO_MODE = "Auto";
+var L1_VARIABLE_DICE_MODE = "Variable Dice";
+var L1_VARIABLE_PLAYERS_MODE = "Variable Players";
+
+// modes to faciliate running of games
+var L2_ROLL_DICE = "Roll Dice Mode";
+var L2_ROLL_DICE_AUTO = "Roll and Auto Pick";
+var L2_CHOOSE_ORDER = "Choose Order Mode";
+var L2_DECIDE_WINNER = "Decide Winner";
+var L2_CHOOSE_NUM_DICE = "Choose number of dice";
+
+// setting initial set up
+var arrayRolls = [];
+var playerOneNum = 0;
+var playerTwoNum = 0;
+var arrayForNums = [];
+var currentPlayer = 1;
+var numOfPlayers = 2;
+var numofDice = 2;
+var L2GameMode = L2_ROLL_DICE;
+var L1GameMode = L1_PICK_GAME_MODE;
+var playerOneScore = 0;
+var playerTwoScore = 0;
+
+// ____________________________SUPPORTING FUNCTIONS_______________________________
+
+// 2 functions for player to select their order
+var selectAB = function () {
+  number = Number(`${arrayRolls[0]}` + `${arrayRolls[1]}`);
+  return number;
+};
+var selectBA = function () {
+  number = Number(`${arrayRolls[0]}` + `${arrayRolls[1]}`);
+  return number;
+};
+
+// standard dice roll functions
+var diceRoll = function () {
+  var randomNum = Math.floor(Math.random() * 6) + 1;
+  return randomNum;
+};
+
+// function to get the "scores" which is the sum of all the rolls
+// prepared code to be able to calculate for multiple players because the oder in which values are pushed into array is the same as order of player picking their numbers.
+
+var getScores = function (player) {
+  var myOutputValue = 0;
+  // input will help determine whose scores to calculate
+  var counter = player - 1;
+  while (counter < arrayForNums.length) {
+    myOutputValue = myOutputValue + arrayForNums[counter];
+    // this counter will help to only pull the numbers of the specific player from the array
+    counter += numOfPlayers;
+  }
+  return myOutputValue;
+};
```

Comment:
Great decomposition. For the naming of myOutputValue although that's what we're very used to, in this context, it's quite different.

Maybe some thing like playerScore would be clearer!